crates/core/component/dao/src/action/dao_deposit.rs

```diff
@@ -1,25 +1,28 @@
+use anyhow::{Context, Error};
+use prost::Message;
+use serde::{Deserialize, Serialize};
 use std::convert::{TryFrom, TryInto};
 
-use anyhow::{Context, Error};
 use penumbra_asset::{Balance, Value};
 use penumbra_chain::{EffectHash, EffectingData};
 use penumbra_proto::{core::governance::v1alpha1 as pb, DomainType, TypeUrl};
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(try_from = "pb::DaoDeposit", into = "pb::DaoDeposit")]
 pub struct DaoDeposit {
     pub value: Value,
 }
 
 impl EffectingData for DaoDeposit {
     fn effect_hash(&self) -> EffectHash {
+        let effecting_data: pb::DaoDeposit = self.clone().into();
+
         let mut state = blake2b_simd::Params::default()
             .personal(b"PAH:daodeposit")
```

Comment:
One thought I had here as I was reading this is that we might be able to remove the personalization strings. However, I don't think we can, because we might have a theoretical scenario where it's possible for two different protos to encode to the same bytes, and we'd want their hashes to be different if they're actually different things, to avoid shenanigans.

I don't think protobufs encode the "name" of the struct anywhere either, which would also avoid this need.

Reply:
What if, instead of using a personalization string, we hashed the `TYPE_URL` provided by the [`TypeUrl` trait implemented for any domain type](https://rustdoc.penumbra.zone/main/penumbra_proto/trait.TypeUrl.html) and then the message encoding?

Reply:
this is a good idea! I've made that change in cdbfac7391fc15770f08e0b347962959c71a5be9, where we first hash the length of the variable-length `TYPE_URL`, then the `TYPE_URL`, and finally the corresponding message